MultiDB support

realityking opened the pull request:
As requested by Mark here's pull with all the changes needed for the CMS's MultiDB work.

hooduku committed:
Improve the support for databases other than MySQL.

eddieajau committed:
SQL server merge fixes.
Added support to store the database name in the JDatabase object.

hooduku committed:
More MultiDB work.

dextercowley commented:
I hope this can be committed soon. We would very much like to get this in before the CMS 2.5 beta (so platform 11.4?). Thanks!

eddieajau commented:
I haven't finished reviewing the work from Sudhi, I was just getting it into a condition where it was reviewable.  Don't merge yet - it's not ready.

joomla-jenkins commented:
The tests completed but there was a problem parsing the report.
Checkstyle analysis reported 208 warnings and 69 errors.

joomla-jenkins commented:
Build triggered by changes to the head.

The tests completed but there was a problem parsing the report.
Checkstyle analysis reported 208 warnings and 69 errors.

joomla-jenkins commented:
Build triggered by changes to the head.

The tests completed but there was a problem parsing the report.
Checkstyle analysis reported 208 warnings and 69 errors.

eddieajau committed:
Clean up MultiDB work:
Add new methods to JDatabaseMock
Fix up odd bug in JTableNested test for the store method.
Add test for storing a new user.
Change db getDateFormat to toSql.

eddieajau commented:
Ok, can you pull from my sudhi-staging branch.  It should be all up to date.

chdemko commented on libraries/joomla/database/table/user.php:
```diff
@@ -465,7 +465,7 @@ public function setLastVisit($timeStamp = null, $userId = null)
 		$query->update($db->quoteName($this->_tbl));
-		$query->set($db->quoteName('lastvisitDate') . '=' . $db->quote($date->format($db->getDateFormat())));
+		$query->set($db->quoteName('lastvisitDate') . '=' . $db->quote($date->format($this->_db->getDateFormat())));
 		$query->where($db->quoteName('id') . '=' . (int) $userId);
```
```
$date->format($this->_db->getDateFormat())
```

could be replaced by

```
$date->toSql()
```

eddieajau replied:
Fixed in my branch.

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1895 tests and 11033 assertions.
Checkstyle analysis reported 200 warnings and 62 errors.

realityking committed:
Fix codestyle issues.

realityking commented:
I pulled in the work from your branch and hopefully fixed all the the code style errors.

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1895 tests and 11033 assertions.
Checkstyle analysis reported 199 warnings and 7 errors.

eddieajau commented:
Errors

libraries/joomla/database/databasequery.php:810
The comments for parameters $table (1) and $increment_field (2) do not align

libraries/joomla/database/databasequery.php:810
Missing comment for param "$increment_field" at position 2

libraries/joomla/database/table/usergroup.php:217
Expected "foreach (...)\n...{\n"; found "foreach(...)\n...{\n"

libraries/joomla/database/table/usergroup.php:231
Concat operator must be followed by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be preceeded by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be followed by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be preceeded by one space

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1962 tests and 11128 assertions.
Checkstyle analysis reported 199 warnings and 3 errors.

eddieajau commented:
I give up.  The hooduku commits put back a heap of stuff that was unnecessary or wrong.  Please revert them.

dextercowley commented:
Hi Andrew. I didn't commit anything to the platform, so I'm not sure
what you are asking me to do. I'm on line if you want to chat. Thanks.
Mark

On Thu, Dec 15, 2011 at 2:03 PM, Andrew Eddie
reply@reply.github.com
wrote:

> I give up.  The hooduku commits put back a heap of stuff that was unnecessary or wrong.  Please revert them.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/joomla/joomla-platform/pull/635#issuecomment-3169850

eddieajau commented:
88d7b8a and 18c9d57 need to be taken out of this branch.  Nothing you did or can do.

elinw commented:
I suggest closing this pull request. If  Andrew is really giving up then we can make a new one just based on Andrew's plus the code style clean up I did in my pull of Andrew's  branch, assuming that nothing works worse with that than what we have now. I think the messiness of  this is more about people not understanding how to work with git than any ill-intent.

eddieajau commented:
Um, let me be clear. "I give up" is meant in terms of exasperation.  I have spent significant spare time on this already but don't have any bandwidth left to sort out whatever has happened this close to Christmas.  Sorry if that doesn't suit but that's the way it is.

elinw commented:
Yeah we all understand that feeling.
